### PR TITLE
Implement socket io & history actions

### PR DESCRIPTION
**Pull request by yana617**

### CODE REVIEW

File: src/middlewares/validations/validateHistoryAction.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    exists: true,
+  },
+  user_from: {
+    in: ['body'],
+    errorMessage: ERRORS.INVALID_USER_ID,
+    isLength: {
+      options: { min: 36, max: 36 },
+    },
+    isString: true,
+    optional: true,
```
Comment (yana617) [inline src/middlewares/validations/validateHistoryAction.js:23]: Remove

File: src/middlewares/validations/validateHistoryAction.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      options: [historyActionTypes],
+    },
+    isString: true,
+    exists: true,
+  },
+  user_from: {
+    in: ['body'],
+    errorMessage: ERRORS.INVALID_USER_ID,
+    isLength: {
+      options: { min: 36, max: 36 },
```
Comment (yana617) [inline src/middlewares/validations/validateHistoryAction.js:20]: isUUID: {
      errorMessage: 'user_id must be a uuid',
    },

File: src/middlewares/validations/validateHistoryAction.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    },
+    isString: true,
+    optional: true,
+  },
+  user_to: {
+    in: ['body'],
+    errorMessage: ERRORS.INVALID_USER_ID,
+    isLength: {
+      options: { min: 36, max: 36 },
```
Comment (yana617) [inline src/middlewares/validations/validateHistoryAction.js:29]: uuid

File: src/middlewares/validations/validateHistoryAction.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    },
+    isString: true,
+    optional: true,
+  },
+  new_role: {
```
Comment (yana617) [inline src/middlewares/validations/validateHistoryAction.js:34]: add guest_to

File: src/middlewares/validations/validateHistoryAction.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  action_type: {
+    in: ['body'],
+    errorMessage: ERRORS.INVALID_ACTION_TYPE,
+    isIn: {
+      options: [historyActionTypes],
+    },
+    isString: true,
+    exists: true,
+  },
+  user_from: {
```
Comment (yana617) [inline src/middlewares/validations/validateHistoryAction.js:16]: user_from_id?

File: src/tests/historyAction.test.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    const response = await request(app)
+      .get('/history-actions')
+      .expect(200);
+
+    const { data: { historyActions, total } } = response.body;
+    expect(historyActions.length).toBe(0);
+    expect(total).toBe(0);
+  });
+
+  test('Should fail with not enough permissions for guest creation', async () => {
```
Comment (yana617) [inline src/tests/historyAction.test.js:61]: edit description

**Review by Aphelmion — APPROVED** [review]

**Pull request merged**